dts/bindings/display/sitronix,st7789h2-st-fmc.yaml

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2022, Jeremy LOCHE
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  ST7789H2 320x240 display controller
+
+  Using the FMC peripheral of STM32 microcontrollers parallel bus
+  interface for the read/write operations to the display.
+
+  You need to configure the FMC peripheral as an SRAM controller
+  with correct bank access timing and configuration.
+
+  This drivers only needs 2 address to work and defined in "reg = <>":
+    Register/Command address:
+      Used to write the LCD controller registers
+    Data address:
+      Used to write LCD RAM display data
+
+  Example configuration:
+
+  &fmc {
+    status = "okay";
+    /* ST7789H2 FMC pinout based on your boards layout */
+    pinctrl-0 = <...>;
+    pinctrl-names = "default";
+
+    /* ST7789H2 FMC SRAM interface*/
+    lcd_fmc_sram: sram {
+      compatible = "st,stm32-fmc-nor-psram";
+      #address-cells = <1>;
+      #size-cells = <0>;
+      label = "STM32_FMC_NOR_PSRAM";
+      status = "okay";
+
+      lcd_data_bus: sram@0 {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        reg = <0>;
+
+        /* Configure the FMC bus */
+        st,control = <STM32_FMC_DATA_ADDRESS_MUX_DISABLE
+                      STM32_FMC_MEMORY_TYPE_SRAM
+                      STM32_FMC_NORSRAM_MEM_BUS_WIDTH_16
+                      STM32_FMC_BURST_ACCESS_MODE_DISABLE
+                      STM32_FMC_WAIT_SIGNAL_POLARITY_LOW
+                      STM32_FMC_WAIT_TIMING_BEFORE_WS
+                      STM32_FMC_WRITE_OPERATION_ENABLE
+                      STM32_FMC_WAIT_SIGNAL_DISABLE
+                      STM32_FMC_EXTENDED_MODE_DISABLE
+                      STM32_FMC_ASYNCHRONOUS_WAIT_DISABLE
+                      STM32_FMC_WRITE_BURST_DISABLE
+                      STM32_FMC_CONTINUOUS_CLOCK_SYNC_ONLY
+                      STM32_FMC_WRITE_FIFO_DISABLE
+                      STM32_FMC_PAGE_SIZE_NONE>;
+
+        st,timing = <1 1 32 0 2 2 STM32_FMC_ACCESS_MODE_A>;
+        st,timing-ext = <5 1 3 2 STM32_FMC_ACCESS_MODE_A>;
+      };
+    };
+  };
+
+  &lcd_data_bus {
+    st7789h2: st7789h2@0 {
+      /* CMD reg */
+      /* DATA reg */
+      reg = <0x60000000
+            0x60000002>;
+
+      compatible = "sitronix,st7789h2-st-fmc";
```

Comment:
@lochej That's a great work, but wouldn't it be possible to push this one step further and completely split st7792h2 driver from STM32 FMC ?

Reply:
Agreed, I had a similar complaint when a panel driver was introduced with vendor dependent MIPI-DSI code inside. Fortunately it lead to the experimental MIPI-DSI interface, so that panel drivers stay vendor independent. Something similar should be done here.

Reply:
Great, I'd love to go in this direction. Could you help me find out how the splitting should be done ? 
My driver implementation doesn't use any FMC vendor specific code, it just writes at a given address pointer in the reg properties. 
So implementation wise, it is more or less split from the FMC driver (except from the name ?). 

Are there any ideas for improving that ? Creating an a memory bus interface/driver that could be parallel bitbang or through FMC or anything else ?

Reply:
@lochej, I guess @gmarull refers to #36875 where [include/zephyr/drivers/mipi_dsi.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/mipi_dsi.h) was introduced. 

However, I am not sure if FMC approach is something generic across several vendors. Therefore, it is hard to justify another read/write abstraction.

Seems like NXP has [FlexIO ](https://www.nxp.com/docs/en/application-note/AN5313.pdf) that can be used "Emulate parallel communication protocol". But I didn't look into it.